pkg/controller/metadata/metadata_controller_test.go

```diff
@@ -558,3 +558,48 @@ func generateCertChain(t *testing.T, ctx context.Context, c client.Client, g *Go
 
 	return rootCertReq, intCertReq, metaCertReq, tearDown
 }
+
+func TestShouldRegenerate(t *testing.T) {
+	g := NewGomegaWithT(t)
+
+	const ConstantHash = "Im a constant hash"
+
+	mockSecrets := corev1.Secret{}
+	mockSecrets.ObjectMeta.Annotations = make(map[string]string)
+	mockSecrets.Data = make(map[string][]byte)
+
+	// Hashes should differ, so should be true to regenerate
+	mockSecrets.ObjectMeta.Annotations[versionAnnotation] = ""
+	g.Eventually(ShouldRegenerate(&mockSecrets, ConstantHash)).Should(BeTrue())
+
+	// Hash should now match, but there is no data for the expiration, this simulates a upgrade.
+	mockSecrets.ObjectMeta.Annotations[versionAnnotation] = ConstantHash
+	g.Eventually(ShouldRegenerate(&mockSecrets, ConstantHash)).Should(BeTrue())
+
+	// There should be a parse error.
+	mockSecrets.Data[validityDays] = []byte("30")
+	mockSecrets.Data[validUntil] = []byte("")
+	g.Eventually(ShouldRegenerate(&mockSecrets, ConstantHash)).Should(BeTrue())
+
+	// Should regenerate as time is in the past.
+	mockSecrets.Data[validUntil] = []byte("Mon, 02 Jan 2006 15:04:05 -0700")
+	g.Eventually(ShouldRegenerate(&mockSecrets, ConstantHash)).Should(BeTrue())
+
+	// Should regenerate if half of the validity days.
+	mockSecrets.Data[validUntil] = []byte(time.Now().AddDate(0, 0, 14).Format(time.RFC1123Z))
```

Comment:
```suggestion
	mockSecrets.Data[validUntil] = []byte(time.Now().AddDate(0, 0, 15).Format(time.RFC1123Z))
```
15 is the exact half-life so we should test regeneration gets triggered at this point.